pkg/queue/postgres/setup.go

```diff
@@ -361,22 +372,37 @@ func (l indexList) generateStatements() []string {
 	indexDefinitions := make([]string, 0, len(l))
 	for _, index := range l {
 		stmt := strings.Builder{}
-		stmt.WriteString("CREATE INDEX IF NOT EXISTS ")
+		stmt.WriteString("CREATE ")
+		if index.Unique {
+			stmt.WriteString("UNIQUE ")
+		}
+		stmt.WriteString("INDEX IF NOT EXISTS ")
 
 		columnList := strings.Join(index.Columns, ",")
 
-		stmt.WriteString(fmt.Sprintf(
-			"%s_%s_idx ON %s",
-			index.Table,
-			cstrings.ToUnderscoreCase(columnList),
-			index.Table,
-		))
+		// is a noop when string is empty
+		stmt.WriteString(index.Name)
+		if index.Name == "" {
+			stmt.WriteString(fmt.Sprintf(
+				"%s_%s_idx",
+				index.Table,
+				cstrings.ToUnderscoreCase(columnList),
+			))
+		}
```

Comment:
I don't think we should allow indexes without names, the whole thing relies on indexes to have names.

Reply:
they do have names, either you can override it _or_ it is generated from the table and columns

Reply:
well, nobody guarantees the generated name would much what we generate. I'm fine with leaving this check but I personally would rather return an error if the name is not set.